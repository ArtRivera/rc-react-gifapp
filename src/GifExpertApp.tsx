import React,{useState} from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}: {defaultCategories?: string[]}) => {

    const [category, setCategory] = useState(defaultCategories);

    return (
        <>
            <h2>Arthur's Gif App</h2>
            <AddCategory setCategory={setCategory}/>
            <hr/>
            <ol>
                {category.map(categoria => <GifGrid key={categoria} category={categoria}/>)}
            </ol>
        </>
    )
}

export default GifExpertApp
