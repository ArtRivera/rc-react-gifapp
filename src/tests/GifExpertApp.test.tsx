import './setupTests';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpert App Tests', () => {

    test('should match to snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('should mostrar una lista de categorias', () => {
        const categories = ['Dragon', 'WandaVision'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
    
    
})
