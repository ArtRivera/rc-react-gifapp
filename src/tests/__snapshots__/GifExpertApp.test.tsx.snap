// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpert App Tests should match to snapshot 1`] = `
<Fragment>
  <h2>
    Arthur's Gif App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpert App Tests should mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Arthur's Gif App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon"
      key="Dragon"
    />
    <GifGrid
      category="WandaVision"
      key="WandaVision"
    />
  </ol>
</Fragment>
`;
